### ISSUE TITLE
delete() does nothing. It should have a similar behaviour to signOut()

### ISSUE THREAD

**Issue by anasqadrei** [atn832/firebase_auth_mocks#106]
`User.delete()` should make the current user null and notify its listers. i.e. Something similar to `MockFirebaseAuth.signOut()`. Currently it does nothing.

**Comment by atn832** [comment]
You're right, they haven't been implemented yet. If someone is willing to contribute a PR, I'll review and publish the update.
